script/tool/lib/src/common/plugin_utils.dart

```diff
@@ -16,6 +16,25 @@ enum PlatformSupport {
   federated,
 }
 
+/// Returns true if [package] is a Flutter plugin.
+bool isFlutterPlugin(Directory package) {
+  try {
+    final File pubspecFile = package.childFile('pubspec.yaml');
+    final YamlMap pubspecYaml =
+        loadYaml(pubspecFile.readAsStringSync()) as YamlMap;
+    final YamlMap? flutterSection = pubspecYaml['flutter'] as YamlMap?;
+    if (flutterSection == null) {
+      return false;
+    }
+    final YamlMap? pluginSection = flutterSection['plugin'] as YamlMap?;
+    return pluginSection != null;
+  } on FileSystemException {
+    return false;
+  } on YamlException {
```

Comment:
Wouldn't it be better to use explicit checks instead of relying on exceptions?  It seems like a malformed yaml file would result in a `isFlutterPlugin == false` instead of crashing the tool which I would have expected.

Reply:
> Wouldn't it be better to use explicit checks instead of relying on exceptions?

Removed the `FileSystemException` in favor of an explicit check. (I'm not sure why the code was originally written that way; this was just a move refactor).

> It seems like a malformed yaml file would result in a `isFlutterPlugin == false` instead of crashing the tool which I would have expected.

In practice that wouldn't actually be a problem since we have a different CI test that validates the pubspecs, but I agree we shouldn't rely on it (and in fact, that's much more recent than this pubspec parsing code). I removed the exception handler for `YamlException`, and I've added a catch-all to `PackageLoopingCommand`'s `runForPackage`. That covers most of the code the tool runs at this point, and is a good idea to add in general since a central goal of that command base class is to always keep going, and then report all errors at the end.